<a href="https://colab.research.google.com/github/yasirabd/solver-society-job-data/blob/main/1_Eksplore_data_Jobstreet_Glints.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Objective
Melakukan eksplorasi data Jobstreet dan Glints.

Data yang dibutuhkan:
1. jobstreet_4oct.csv
2. glints_4_oct.csv

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import re

# Jobstreet

In [ ]:
jobstreet = pd.read_csv("/content/drive/My Drive/Data Loker/jobstreet_4oct.csv")
jobstreet.head()

platform  ...                                   company_overview
0  jobstreet  ...  [SEP][SEP] PT. Axindo Infotama didirikan di Ja...
1  jobstreet  ...  [SEP]We are an international market research c...
2  jobstreet  ...  [SEP]OTO Group is a subsidiary company of SUMI...
3  jobstreet  ...  [SEP]Kami menghimbau agar Anda berhati-hati sa...
4  jobstreet  ...  [SEP]OTO Group is a subsidiary company of SUMI...

[5 rows x 18 columns]

In [ ]:
jobstreet.shape

(14150, 18)

In [ ]:
jobstreet.describe().T

count  ...   freq
platform                          14150  ...  14150
job_position                      14141  ...     60
company_name                      14141  ...    548
years_of_experience               14137  ...   3417
company_location                  12720  ...   1786
company_address                   11290  ...    114
posting_date                      14141  ...    782
closing_date                      14141  ...    764
job_description                   14141  ...     10
average_processing_time           13171  ...  10089
company_industry                  13593  ...   1712
company_site                      10242  ...    142
company_size                      13579  ...   3578
work_environment_waktu_bekerja    12519  ...   7999
work_environment_gaya_berpakaian  12430  ...   7042
work_environment_tunjangan        11426  ...   2812
work_environment_bahasa           12491  ...   9215
company_overview                  14141  ...    139

[18 rows x 4 columns]

In [ ]:
# check null value

jobstreet.isnull().sum()

platform                               0
job_position                           9
company_name                           9
years_of_experience                   13
company_location                    1430
company_address                     2860
posting_date                           9
closing_date                           9
job_description                        9
average_processing_time              979
company_industry                     557
company_site                        3908
company_size                         571
work_environment_waktu_bekerja      1631
work_environment_gaya_berpakaian    1720
work_environment_tunjangan          2724
work_environment_bahasa             1659
company_overview                       9
dtype: int64

> Banyak terdapat data kosong, perlu cleaning data.

Kita coba, lakukan eksplorasi sederhana tiap parameter pada Jobstreet.

| No | Attributes                       | Keterangan                                                  |
|----|----------------------------------|-------------------------------------------------------------|
| 1  | platform                         | Tempat data diambil (jobstreet)                             |
| 2  | job_position                     | Posisi pekerjaan yang dicari perusahaan                     |
| 3  | company_name                     | Nama perusahaan yang membuka lowongan kerja                 |
| 4  | years_of_experience              | Lama pengalaman yang dibutuhkan untuk apply pekerjaan       |
| 5  | company_location                 | Lokasi perusahaan yang membuka lowongan kerja               |
| 6  | company_address                  | Alamat perusahaan                                           |
| 7  | posting_date                     | Waktu perusahaan posting lowongan kerja pada platform       |
| 8  | closing_date                     | Waktu post lowongan kerja selesai ditampilkan pada platform |
| 9  | job_description                  | Deskripsi dari kualifikasi dan tanggung jawab pekerjaan     |
| 10 | average_processing_time          | Lama waktu perusahaan memproses lamaran kerja               |
| 11 | company_industry                 | Jenis industri perusahaan                                   |
| 12 | company_site                     | Website perusahaan                                          |
| 13 | company_size                     | Ukuran perusahaan dari jumlah karyawan                      |
| 14 | work_environment_waktu_bekerja   | Waktu bekerja karyawan                                      |
| 15 | work_environment_gaya_berpakaian | Pakaian yang digunakan karyawan formal/non-formal           |
| 16 | work_environment_tunjangan       | Tunjangan yang diberikan pada karyawan                      |
| 17 | work_environment_bahasa          | Bahasa yang digunakan pada perusahaan                       |
| 18 | company_overview                 | Deskripsi perusahaan                                        |

## job_position

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_position = jobstreet['job_position'].str.lower().fillna('kosong')

job_position.value_counts()

sales executive                                    80
accounting staff                                   68
digital marketing                                  56
marketing                                          49
sales manager                                      48
                                                   ..
marketing head (parung)                             1
assistant sales staff (yl palembang)                1
(jv1912003) account officer - sales - pekanbaru     1
seo content specialist                              1
project director (id: 501476)                       1
Name: job_position, Length: 10078, dtype: int64

**Sales Executive** merupakan job_position yang paling banyak jumlah kemunculan pada data.

Jika kita perhatikan data unique pada job_position adalah **10078** dari total data kita **14150**. Artinya data job_position masih kotor dan perlu cleaning.

Kita coba cek kemunculan jenis pekerjaan **sales executive** pada data.

In [ ]:
# check banyak jenis sales executive
temp = list()
for pos in job_position:
    if 'sales executive' in pos:
        temp.append(pos)
pd.Series(temp).unique()

array(['sales executive (mandarin speaking)', 'sales executive',
       'sales executive (cirebon)', 'sales executive (pontianak)',
       'sales executive veteriner', 'sales executive (pekanbaru)',
       'sales executive supplement division',
       'sales executive horeka & b2b semarang',
       'sales executive horeka & b2b - yogyakarta',
       'sales executive horeka & b2b jawa barat - bandung',
       'sales executive (penempatan berau)',
       'sales executive - enclosure, rack & cooling unit mnc',
       'sales executive (financial services)',
       'sales executive new channel',
       'sales executive - food service (hotel, restaurant, catering) balikpapan branch',
       'sales executive ( penempatan depok )', 'sales executive solo',
       'technical sales executive (jabodetabek & semarang)',
       'sales executive pontianak', 'sales executive it',
       'it sales field/sales executive',
       'sales executive - personal care (bogor)',
       'senior sales executive',

> Terlihat sales executive sendiri memiliki beragam jenis dengan penjelas. Perlu kita bersihkan data penjelas. 

Untuk membersihkan data, kita dapat menggunakan regex.

In [ ]:
a = "sales executive (education consultant)"

regex = re.compile(".*?\((.*?)\)")
re.sub("[\(\[].*?[\)\]]", "", a).strip()

'sales executive'

## company_name

In [ ]:
# check null value
jobstreet['company_name'].isnull().sum()

9

In [ ]:
# fillna dengan string 'kosong'
company_name = jobstreet['company_name'].fillna('kosong')

company_name.value_counts()

Perusahaan Dirahasiakan                                       548
OTO Group                                                     142
PT Michael Page Internasional Indonesia (Recruitment Firm)     83
PT AXA Mandiri Financial Services                              62
PT. Aurora Group (Recruitment Firm)                            62
                                                             ... 
PT Bina Persada Jaya                                            1
Century 21 Indonesia                                            1
PT Masjidpedia Inovasi Bersama                                  1
PT Tolan Tiga Indonesia                                         1
PT Alistor Jaya Mandiri                                         1
Name: company_name, Length: 5107, dtype: int64

> gakada masalah

## years_of_experience

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
yoe = jobstreet['years_of_experience'].str.lower().fillna('kosong')

yoe.value_counts()[:60]

min 1 tahun (pegawai (non-manajemen & non-supervisor))             3417
min 2 tahun (pegawai (non-manajemen & non-supervisor))             2464
min 3 tahun (pegawai (non-manajemen & non-supervisor))             1064
min 2 tahun (supervisor/koordinator)                                936
min 3 tahun (supervisor/koordinator)                                907
min 5 tahun (manajer/asisten manajer)                               839
lulusan baru/pengalaman kerja kurang dari 1 tahun                   791
min 3 tahun (manajer/asisten manajer)                               522
min 5 tahun (supervisor/koordinator)                                453
min 1 tahun (supervisor/koordinator)                                370
min 2 tahun (manajer/asisten manajer)                               310
min 5 tahun (pegawai (non-manajemen & non-supervisor))              306
min 1 tahun (lulusan baru/pengalaman kerja kurang dari 1 tahun)     265
min 4 tahun (supervisor/koordinator)                            

Kita tidak perlu data posisi pada bracket ()

In [ ]:
"min 8 tahun (ceo/gm/direktur/manajer senior)".split('(')[0].strip()

'min 8 tahun'

In [ ]:
result = yoe.values
for idx, years in enumerate(yoe):
    result[idx] = years.split('(')[0].strip()

In [ ]:
pd.Series(result).value_counts()

min 1 tahun                                          4154
min 2 tahun                                          3766
min 3 tahun                                          2529
min 5 tahun                                          1675
lulusan baru/pengalaman kerja kurang dari 1 tahun     791
min 4 tahun                                           443
min 10 tahun                                          269
min 8 tahun                                           150
min 7 tahun                                           137
min 6 tahun                                           100
min 15 tahun                                           63
min 12 tahun                                           24
kosong                                                 13
min 9 tahun                                             9
min 13 tahun                                            4
supervisor/koordinator                                  4
pegawai                                                 3
lebih dari 24 

## company_location

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
company_location = jobstreet['company_location'].str.lower().fillna('kosong')

company_location.value_counts()[:60]

indonesia - jakarta raya                                         1786
kosong                                                           1430
indonesia - jakarta raya - jakarta selatan                        689
indonesia - jawa timur - surabaya                                 419
indonesia - jakarta raya - jakarta barat                          395
indonesia - jakarta raya - jakarta pusat                          378
indonesia - banten - tangerang                                    357
indonesia - jawa barat - bandung                                  255
indonesia - jakarta raya - jakarta utara                          207
indonesia - jawa tengah - semarang                                183
indonesia - jawa barat - cikarang                                 125
indonesia - jakarta raya - jakarta timur                          113
indonesia - yogyakarta                                            112
indonesia - jawa barat - bekasi                                   110
indonesia - jakarta 

perlu di ekstrak hanya untuk kota/province saja

## company_address
tidak digunakan, karena kurang berguna

## posting_date

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
posting_date = jobstreet['posting_date'].str.lower().fillna('kosong')

posting_date.value_counts()

01-october-2020      782
02-october-2020      726
28-september-2020    716
24-september-2020    708
30-september-2020    685
29-september-2020    682
22-september-2020    678
25-september-2020    665
07-september-2020    663
21-september-2020    602
23-september-2020    600
09-september-2020    595
08-september-2020    592
04-september-2020    582
14-september-2020    575
15-september-2020    560
16-september-2020    560
17-september-2020    543
18-september-2020    525
11-september-2020    468
10-september-2020    402
03-october-2020      228
26-september-2020    184
19-september-2020    158
27-september-2020    134
20-september-2020    113
04-october-2020      110
05-september-2020    109
12-september-2020    106
13-september-2020     50
06-september-2020     40
kosong                 9
Name: posting_date, dtype: int64

> data oke

## closing_date

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
closing_date = jobstreet['closing_date'].str.lower().fillna('kosong')

closing_date.value_counts()

tutup pada 22-october-2020                        764
tutup pada 3 hari | tutup pada 07-october-2020    734
tutup pada 24-october-2020                        696
tutup pada 5 hari | tutup pada 09-october-2020    686
tutup pada 4 hari | tutup pada 08-october-2020    677
tutup pada 28-october-2020                        667
tutup pada 29-october-2020                        664
tutup pada 31-october-2020                        660
tutup pada 04-october-2020                        646
tutup pada 21-october-2020                        646
tutup pada 30-october-2020                        640
tutup pada 16-october-2020                        629
tutup pada 15-october-2020                        622
tutup pada 14-october-2020                        617
tutup pada 23-october-2020                        608
tutup pada 17-october-2020                        598
tutup pada 01-november-2020                       591
tutup pada 25-october-2020                        588
tutup pada 18-october-2020  

> Kita perlu ambil tanggal saja

In [ ]:
result = closing_date.values

for idx, date_close in enumerate(result):
    if "|" in date_close:
        result[idx] = date_close.split('|')[1].split()[-1].strip()
    else:
        result[idx] = date_close.split()[-1].strip()

In [ ]:
pd.Series(result).value_counts()

22-october-2020     764
07-october-2020     747
24-october-2020     696
09-october-2020     694
08-october-2020     684
28-october-2020     667
29-october-2020     664
31-october-2020     660
04-october-2020     648
21-october-2020     646
30-october-2020     640
16-october-2020     629
15-october-2020     622
14-october-2020     617
23-october-2020     608
17-october-2020     598
01-november-2020    591
25-october-2020     588
18-october-2020     563
11-october-2020     535
10-october-2020     463
02-november-2020    132
05-october-2020     127
19-october-2020     127
12-october-2020     110
26-october-2020     110
13-october-2020      56
27-october-2020      54
06-october-2020      48
20-october-2020      47
kosong                9
03-november-2020      6
dtype: int64

## job_description

Kita perlu ekstraksi:
1. pendidikan
2. skill

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_description = jobstreet['job_description'].str.lower().fillna('kosong')

job_description.head()

0    [sep][sep]usia max. 35 tahun[sep]pendidikan mi...
1    responsibilities:[sep]brief description of dut...
2    [sep]lingkup pekerjaan: [sep]menjaga account d...
3    [sep]candidate must possess at least diploma i...
4    [sep]lingkup pekerjaan: [sep]menjaga account d...
Name: job_description, dtype: object

In [ ]:
# cari job description yang mengandung 
# kata kunci 'pendidikan', 'degree', 'lulusan'
result = job_description.values

count = 0
for idx, job_desc in enumerate(result):
    if 'pendidikan' in job_desc:
        count += 1
    elif 'degree' in job_desc:
        count += 1
    elif 'lulusan' in job_desc:
        count += 1
count

9698

Coba lihat pekerjaaan yang mencantumkan jenis kelamin.

In [ ]:
# cari job description yang mengandung
result = job_description.values

count = 0
for idx, job_desc in enumerate(result):
    if 'laki-laki' in job_desc:
        count += 1
    elif 'pria' in job_desc:
        count += 1
    elif 'wanita' in job_desc:
        count += 1
    elif 'perempuan' in job_desc:
        count += 1
    elif 'woman' in job_desc:
        count += 1
count

317

Pada data job_description, terdapat 2 bahasa
1. Inggris
2. Indonesia

Kita perlu deteksi bahasa dulu sebelum melakukan ekstraksi.

In [ ]:
job_description[3]

'[sep]candidate must possess at least diploma in nursing or equivalent.[sep]at least 1 year(s) of working experience in the related field is required for this position.[sep]required skill(s): nurse[sep]preferably staff (non-management & non-supervisor) specialized in healthcare - nurse/medical support & assistant or equivalent.[sep]'

In [ ]:
!pip install spacy-langdetect -q

Coba deteksi text pada beberapa text sample.

In [ ]:
import spacy
from spacy_langdetect import LanguageDetector
nlp = spacy.load("en")

nlp.add_pipe(LanguageDetector(), name="language_detector", last=True)

# text = "This is English text. Er lebt mit seinen Eltern und seiner Schwester in Berlin. Yo me divierto todos los días en el parque. Je m'appelle Angélica Summer, j'ai 12 ans et je suis canadienne."
# text = "[sep]candidate must possess at least diploma in nursing or equivalent.[sep]at least 1 year(s) of working experience in the related field is required for this position.[sep]required skill(s): nurse[sep]preferably staff (non-management & non-supervisor) specialized in healthcare - nurse/medical support & assistant or equivalent."
# text = "saya suka makan nasi goreng"
text = "[sep]lingkup pekerjaan: [sep]menjaga account dan memberikan solusi dalam penyelesaian pembayaran angsuran customer[sep] [sep]kualifikasi:[sep][sep] usia maksimal 28 tahun[sep] pendidikan d3/s1 (slta dipertimbangkan)[sep] memiliki kemampuan negosiasi yang baik[sep] dinamis dan komunikatif[sep] menyukai pekerjaan lapangan[sep] memiliki sepeda motor dan sim c[sep][sep]kompensasi & benefit:[sep][sep] gaji pokok[sep] tunjangan tetap[sep] bpjs ketenagakerjaan dan kesehatan[sep] insentif[sep] thr dan bonus[sep][sep]lokasi kerja : [sep]pati[sep] [sep]daftar, lamar, serta tes online posisi tersebut di www.karir.oto.co.id.[sep]"

doc = nlp(text)
# document level language detection. Think of it like average language of document!
print(doc._.language)

{'language': 'id', 'score': 0.9999983578060843}


> spacy_lang dapat mendeteksi bahasa indonesia dan inggris

## average_processing_time

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
average_processing_time = jobstreet['average_processing_time'].str.lower().fillna('kosong')

average_processing_time.value_counts()

lebih dari 2 minggu    10089
kosong                   979
1 hari                   720
2 hari                   279
6 hari                   251
11 hari                  213
3 hari                   213
10 hari                  202
4 hari                   194
5 hari                   191
12 hari                  189
13 hari                  177
8 hari                   173
9 hari                   147
7 hari                   133
Name: average_processing_time, dtype: int64

> Kita perlu menyamakan satuan menjadi **hari** semua serta cukup ambil numeric angka.

## company_industry	

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
company_industry = jobstreet['company_industry'].str.lower().fillna('kosong')

company_industry.value_counts()

manufaktur/produksi                                     1712
perbankan/pelayanan keuangan                             883
komputer/teknik informatika (perangkat lunak)            726
umum & grosir                                            703
retail/merchandise                                       691
produk konsumen/barang konsumen yang bergerak cepat      621
manajemen/konsulting hr                                  616
kosong                                                   557
transportasi/logistik                                    535
properti/real estate                                     517
makanan & minuman/katering/restoran                      485
konstruksi/bangunan/teknik                               482
kesehatan/medis                                          476
agrikultural/perkebunan/peternakan unggas/perikanan      397
konsultasi (it, ilmu pengetahuan, teknis & teknikal)     381
asuransi                                                 360
telekomunikasi          

> Kita bisa klasifikasikan menjadi lebih sederhana berdasarkan [Klasifikasi Baku Lapangan Usaha Indonesia 2020](https://infiniti.id/kbli)

## company_site
tidak dipakai

## company_size

In [ ]:
# convert to lower case dan fillna dengan string 'tidak spesifik'
company_size = jobstreet['company_size'].str.lower().fillna('tidak spesifik')

company_size.value_counts()

51 - 200 pekerja           3578
1- 50 pekerja              2751
201 - 500 pekerja          2165
501 - 1000 pekerja         1511
1001 - 2000 pekerja        1231
lebih dari 5000 pekerja    1161
2001 - 5000 pekerja         945
tidak spesifik              808
Name: company_size, dtype: int64

> Data cukup bagus, ter-cluster.

## work_environment_waktu_bekerja

In [ ]:
# convert to lower case dan fillna dengan string 'tidak spesifik'
work_environment_waktu_bekerja = jobstreet['work_environment_waktu_bekerja'].str.lower().fillna('tidak spesifik')

work_environment_waktu_bekerja.value_counts()

waktu regular, senin - jumat            7999
tidak spesifik                          1631
sabtu atau memerlukan shift             1130
senin - sabtu                            355
monday - saturday                        269
                                        ... 
mondays - fridays/saturdays                1
monday to saturday 8 am - 4 pm             1
target based                               1
regular, saturdays or shift required       1
senin - minggu, jam 08.00 - 20.00          1
Name: work_environment_waktu_bekerja, Length: 678, dtype: int64

> Kita bisa lakukan cleaning dan cluster waktu bekerja.

## work_environment_gaya_berpakaian

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
work_environment_gaya_berpakaian = jobstreet['work_environment_gaya_berpakaian'].str.lower().fillna('kosong')

work_environment_gaya_berpakaian.value_counts()

bisnis (contoh: kemeja)                                                7042
kasual (contoh: kaos)                                                  2301
kosong                                                                 1720
formil (contoh: kemeja + dasi)                                         1546
uniform                                                                 308
                                                                       ... 
company’s dress code                                                      1
formal in terms of meeting with other parties, casual in day to day       1
formal & casual                                                           1
casual (t-shirts, polo shirts, shirts)                                    1
menyesuaikan                                                              1
Name: work_environment_gaya_berpakaian, Length: 269, dtype: int64

> Kita perlu cleaning data dengan pengelompokkan jenis gaya berpakaian.

## work_environment_tunjangan

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
work_environment_tunjangan = jobstreet['work_environment_tunjangan'].str.lower().fillna('kosong')

work_environment_tunjangan.value_counts()

kesehatan                                 2816
kosong                                    2724
kesehatan, tunjangan lainnya               956
kesehatan, parkir                          571
tunjangan lainnya                          543
                                          ... 
sales bonus                                  1
kesehatan, tunjangan lainnya, asuransi       1
kesehatan, dplk (pension fund)               1
tunjangan kesehatan, tunjangan lainnya       1
makan, tempat tinggal                        1
Name: work_environment_tunjangan, Length: 1081, dtype: int64

> Kita bisa lakukan cluster tunjangan

## work_environment_bahasa

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
work_environment_bahasa = jobstreet['work_environment_bahasa'].str.lower().fillna('kosong')

work_environment_bahasa.value_counts()[:50]

bahasa indonesia                                     9215
bahasa inggris                                       1868
kosong                                               1659
bahasa indonesia & english                            208
bahasa indonesia - english                            162
bahasa indonesia and english                          140
english & bahasa indonesia                             46
bahasa indonesia dan english                           43
bahasa indonesia, english                              42
mandarin                                               38
bahasa indonesia dan inggris                           31
english & bahasa                                       28
english and bahasa indonesia                           27
indonesia - english                                    24
bahasa indonesia & bahasa inggris                      24
english-bahasa indonesia                               24
indonesia & english                                    22
bahasa indones

> Kita perlu cleaning data dengan menyamakan kategori bahasa pada data.

## company_overview
tidak dipakai

# Glints

In [ ]:
glints = pd.read_csv("/content/drive/My Drive/Data Loker/glints_4oct.csv")
glints.head()

platform  ...                                     skills_to_have
0   glints  ...  Must Have Skills [SEP] .NET [SEP] English [SEP...
1   glints  ...  Must Have Skills [SEP] ReactJS [SEP] HTML [SEP...
2   glints  ...  Must Have Skills [SEP] Teamwork [SEP] Microsof...
3   glints  ...  Must Have Skills [SEP] Agile Scrum [SEP] Engli...
4   glints  ...  Must Have Skills [SEP] Cms Wordpress [SEP] Wor...

[5 rows x 9 columns]

In [ ]:
glints.describe().T

count  ...  freq
platform          1950  ...  1950
job_position      1950  ...    24
company_name      1950  ...    30
job_category      1950  ...   609
company_location  1950  ...  1198
job_salary        1950  ...  1368
job_type          1950  ...  1723
job_desc          1950  ...    15
skills_to_have    1950  ...    12

[9 rows x 4 columns]

In [ ]:
# check null values
glints.isnull().sum()

platform            0
job_position        0
company_name        0
job_category        0
company_location    0
job_salary          0
job_type            0
job_desc            0
skills_to_have      0
dtype: int64

## job_position

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_position = glints['job_position'].str.lower()

job_position.value_counts()

backend developer                            25
fullstack developer                          15
frontend developer                           13
account executive                            13
product manager                              12
                                             ..
warehouse manager                             1
accounting and tax specialist                 1
it project ops.                               1
project engineer                              1
customer service freelance (bekasi based)     1
Name: job_position, Length: 1433, dtype: int64

> Bisa kita cleaning agar data lebih distinct

## company_name

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
company_name = glints['company_name']

company_name.value_counts()

PT Paragon Pratama Teknologi (StickEarn)          30
PT. Bintang Jaya Unggul Sentosa                   23
PT Hokinda Citralestari                           20
AIA Singapore Pte Ltd                             20
Confidential Company Talenthunt 1 -006            20
                                                  ..
Newcastle Institute (nuansa Edukasi Indonesia)     1
Thinktive                                          1
PT Bersama Zatta Jaya (elcorps)                    1
Pt Vaganza Solusi Internasional                    1
Urbanetic Pte Ltd                                  1
Name: company_name, Length: 746, dtype: int64

> Cukup bagus

## job_category

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_category = glints['job_category'].str.lower()

job_category.value_counts()

software engineering            609
business development / sales    294
marketing                       209
design                          135
finance                          97
operations                       90
other                            75
administrative                   74
product management               68
media & communications           61
engineering                      56
education                        55
data science                     40
human resource                   33
consulting                       16
legal                            15
management                       15
hospitality and travel            4
healthcare                        3
language and translation          1
Name: job_category, dtype: int64

> Sudah cukup bagus klasifikasi data.

## company_location

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
company_location = glints['company_location'].str.lower()

company_location.value_counts()

jakarta, indonesia            1198
batam, indonesia               147
bandung, indonesia             123
yogyakarta, indonesia           88
surabaya, indonesia             72
tangerang, indonesia            63
south tangerang, indonesia      42
malang, indonesia               28
medan, indonesia                24
bekasi, indonesia               23
depok, indonesia                19
bogor, indonesia                17
semarang, indonesia             14
padang, indonesia                8
pekanbaru, indonesia             7
tulungagung, indonesia           6
denpasar, indonesia              6
badung, indonesia                5
makassar, indonesia              5
cirebon, indonesia               4
serpong, indonesia               4
balikpapan, indonesia            4
mojokerto, indonesia             3
solo, indonesia                  3
cikarang, indonesia              3
singaraja, indonesia             2
sidoarjo, indonesia              2
pandeglang, indonesia            2
sentul, indonesia   

> Dapat kita ekstrak menjadi provinsi dan pulau

## job_salary

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_salary = glints['job_salary'].str.lower()

job_salary.value_counts()

perusahaan tidak menampilkan gaji     1368
idr 4,000,000 -6,000,000 / month        22
idr 4,000,000 -5,000,000 / month        18
idr 5,000,000 -10,000,000 / month       15
idr 3,500,000 -5,000,000 / month        15
                                      ... 
idr 4,800,000 -5,000,000 / month         1
idr 10,000,000 -16,000,000 / month       1
idr 3,900,000 -4,500,000 / month         1
idr 5,500,000 -6,000,000 / month         1
idr 2,800,000 -2,850,000 / month         1
Name: job_salary, Length: 247, dtype: int64

> Perlu cleaning

## job_type

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_type = glints['job_type'].str.lower()

job_type.value_counts()

full_time        1723
internship        142
project_based      58
part_time          27
Name: job_type, dtype: int64

> Data oke

## job_desc

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
job_desc = glints['job_desc'].str.lower()

job_desc.head()

0    requirements: [sep] min diploma / degree in in...
1    what is this job about? [sep] mossery is an e-...
2    front-end spec [sep] have experience with at l...
3    job responsibilities: [sep] • designing and de...
4    [sep] wordpress developer [sep] implement prot...
Name: job_desc, dtype: object

> Kita perlu ekstrak untuk mendapatkan data pendidikan

## skills_to_have

In [ ]:
# convert to lower case dan fillna dengan string 'kosong'
skills_to_have = glints['skills_to_have'].str.lower()

skills_to_have.head()

0    must have skills [sep] .net [sep] english [sep...
1    must have skills [sep] reactjs [sep] html [sep...
2    must have skills [sep] teamwork [sep] microsof...
3    must have skills [sep] agile scrum [sep] engli...
4    must have skills [sep] cms wordpress [sep] wor...
Name: skills_to_have, dtype: object

> Perlu kita cleaning dengan menghilangkan 'must have skills' dan 'nice to have skills' pada skill.